### PR TITLE
for artemis profile update refs to 2.9.0 to 2.14.0 (also 7.7)

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

File: amqcfg/profiles/amq_broker/7.7.0/cluster/off.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+
+_defaults:
+  broker_name: amq
+  broker_home: /opt/amq-broker-7.2.0
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/cluster/off.yaml:5]: Find and replace all 7.2.0... or other versions ideally.

File: amqcfg/profiles/amq_broker/7.7.0/cluster/off.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  cluster_connector_protocol: tcp
+  cluster_connector_address: 0.0.0.0
+  cluster_connector_port: 61616
+  log_level_all: INFO
+  cluster_load_balancing: 'OFF'
+
+render:
+  generator_notice: True
+  licenses: True
+  template: artemis/2.9.0
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/cluster/off.yaml:18]: 2.9.0 or maybe higher version?

File: amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+
+# artemis.profile
+{% include 'artemis/2.14.0/_modules/artemis_profile/default.yaml' %}
+
+# users and roles properties xml
+{% include 'artemis/2.14.0/_modules/users_roles_security/admin.yaml' %}
+
+# bootstrap.xml
+{% include 'amq_broker/7.2.0/_modules/bootstrap_xml/allow.yaml' %}
```
Comment (michalxo) [inline amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml:65]: artemis should not contain any file from amq_broker*

File: amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+{% include 'artemis/2.14.0/_modules/broker_xml/acceptors/default.yaml' %}
+
+{% include 'artemis/2.14.0/_modules/users_roles_security/admin_broker_security_settings.yaml' %}
+
+{% include 'artemis/2.14.0/_modules/broker_xml/address_settings/default.yaml' %}
+
+{% include 'artemis/2.14.0/_modules/broker_xml/addresses/default.yaml' %}
+
+{% include 'amq_broker/7.4.0/_modules/broker_xml/store/jdbc_store.yaml' %}
```
Comment (michalxo) [inline amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml:55]: artemis should not contain any file from amq_broker*

File: amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
+---
+
+_defaults:
+  broker_name: amq
+  broker_home: /opt/amq-broker-7.4.0
```
Comment (michalxo) [inline amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml:5]: IDK how you want it, but if deploying artemis, maybe you want to change broker_home to /opt/artemis instead?
My old automation used amq-broker* but imo it should not be there...

File: amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  broker_instance: /opt/amq-broker-7.4.0-i0
```
Comment (michalxo) [inline amqcfg/profiles/artemis/2.14.0/store/jdbc/default.yaml:6]: same here

File: amqcfg/profiles/amq_broker/7.7.0/store/jdbc/default.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
+---
+
+_defaults:
+  broker_name: amq
+  broker_home: /opt/amq-broker-7.4.0
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/store/jdbc/default.yaml:5]: 7.7.0?

File: amqcfg/profiles/amq_broker/7.7.0/store/jdbc/default.yaml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  broker_instance: /opt/amq-broker-7.4.0-i0
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/store/jdbc/default.yaml:6]: 7.7.0?

File: amqcfg/profiles/amq_broker/7.7.0/store/jdbc/default.yaml [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+{% include 'artemis/2.9.0/_modules/broker_xml/journal/default.yaml' %}
+
+  paging_directory: data/paging
+  bindings_directory: data/bindings
+  large_messages_directory: data/large-messages
+  disk_scan_period: 5000
+  max_disk_usage: 90
+#  global_max_size: 2GB
+
+{% include 'artemis/2.9.0/_modules/broker_xml/critical_analyzer_default.yaml' %}
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/store/jdbc/default.yaml:45]: 2.9.0 -> 2.14.0

File: amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml [hunk 10]
```diff
@@ -0,0 +1,62 @@
+---
+
+_defaults:
+  broker_name: amq
+  broker_home: /opt/amq-broker-7.2.0
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml:5]: 7.2.0 -> 7.7.0

File: amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  gssapi_broker_keytab_path: /path/to/my_keytabs/mybroker.keytab
+  gssapi_broker_principal: amqp/mybroker.myrealm.com
+  gssapi_sasl_login_config_scope: krb5loginconfig
+  gssapi_conf_file_path: /path/to/my_krb5.conf
+  jvm_vendor: sun
+
+render:
+  generator_notice: True
+  licenses: True
+  template: artemis/2.9.0
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml:17]: artemis/2.14.0 ?

File: amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+  paging_directory: data/paging
+  bindings_directory: data/bindings
+  large_messages_directory: data/large-messages
+  disk_scan_period: 5000
+  max_disk_usage: 90
+#  global_max_size: 2GB
+
+
+{% include 'artemis/2.9.0/_modules/broker_xml/critical_analyzer_default.yaml' %}
```
Comment (michalxo) [inline amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml:32]: 2.9.0 -> 2.14.0

File: amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml (hunk 11 above)
Comment (gaohoward) [reply amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml:17]: should be 2.13.0 (I'll add it)

File: amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml (hunk 12 above)
Comment (gaohoward) [reply amqcfg/profiles/amq_broker/7.7.0/security/gssapi_ldap/my.realm.yaml:32]: 7.7.0 is using 2.13.0

**Comment by michalxo** [comment]
As there is too many files, I went only through some of them, and they looked fine. You have changed the versions and I think machine could do better search & replace than I can do visually.

I believe you can go ahead with a merge. In case of an issue, we can fix later in future ;)

**Review by michalxo — APPROVED** [review]
LGTM

**Comment by gaohoward** [comment]
Thanks Michal!

File: amqcfg/profiles/artemis/2.13.0/security/gssapi/ssl_tls/default.yaml [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+  ssl_keystore_password: keystorePass
+  ssl_truststore_path: /etc/ssl/amq-truststore.jks
+  ssl_truststore_password: truststorePass
+
+render:
+  generator_notice: True
+  licenses: True
+  template: artemis/2.13.0
+  profile_path: '{{ profile_path }}'
+  profile_path: artemis/2.13.0/ha/replication.yaml
```
Comment (enkeys) [inline amqcfg/profiles/artemis/2.13.0/security/gssapi/ssl_tls/default.yaml:17]: duplicate item

File: amqcfg/profiles/artemis/2.14.0/security/gssapi/ssl_tls/default.yaml [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+  ssl_keystore_password: keystorePass
+  ssl_truststore_path: /etc/ssl/amq-truststore.jks
+  ssl_truststore_password: truststorePass
+
+render:
+  generator_notice: True
+  licenses: True
+  template: artemis/2.14.0
+  profile_path: '{{ profile_path }}'
+  profile_path: artemis/2.14.0/ha/replication.yaml
```
Comment (enkeys) [inline amqcfg/profiles/artemis/2.14.0/security/gssapi/ssl_tls/default.yaml:17]: duplicate item

**Review by enkeys — COMMENTED** [review]
These duplicates need to be fixed.

**Pull request merged**